plasma_framework/test/src/framework/ExitGameController.test.js

```diff
@@ -0,0 +1,268 @@
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            const nonExistingUniqeuePriority = 123;
+            await expectRevert(
+                this.controller.processExits(this.dummyToken, nonExistingUniqeuePriority, 1),
+                "Top unique priority of the queue is not the same as the specified one"
+            );
+        });
+
+        it('processes when the "top unique priority" is set to 0', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            const tx = await this.controller.processExits(this.dummyToken, 0, 1);
+            await expectEvent.inLogs(tx.logs, 'ProcessedExitsNum', {
+                processedNum: new BN(1),
+                token: this.dummyToken
+            });
+        });
+
+        it('processes when the "top unique priority" is set to the exact top one of the queue', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+            const uniquePriority = await this.dummyExitGame.uniqueuPriorityFromEnqueue();
+
+            const tx = await this.controller.processExits(this.dummyToken, uniquePriority, 1);
+            await expectEvent.inLogs(tx.logs, 'ProcessedExitsNum', {
+                processedNum: new BN(1),
+                token: this.dummyToken
+            });
+        });
+
+        it('calls the "processExit" function of the exit processor when processes', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            await this.controller.processExits(this.dummyToken, 0, 1);
+
+            // 'exitIdProcessed' would be set when processExit is called in the DummyExitGame
+            const exitIdProcessed = await this.dummyExitGame.exitIdProcessed();
+            expect(exitIdProcessed).to.be.bignumber.equal(new BN(this.dummyExit.exitId));
+        });
+
+        it('deletes the processed exit from queue after processed', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            await this.controller.processExits(this.dummyToken, 0, 1);
+
+            const priorityQueueAddress = await this.controller.exitsQueues(this.dummyToken);
+            const priorityQueue = await PriorityQueue.at(priorityQueueAddress);
+            expect(await priorityQueue.currentSize()).to.be.bignumber.equal(new BN(0));
+        });
+
+        it('deletes the exit data after processed', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+            const uniquePriority = await this.dummyExitGame.uniqueuPriorityFromEnqueue();
+
+            await this.controller.processExits(this.dummyToken, 0, 1);
+
+            const exit = await this.controller.exits(uniquePriority);
+            expect(exit.exitProcessor).to.equal(constants.ZERO_ADDRESS);
```

Comment:
If I remember correctly, for reason of dependency between standard and in-flight exits we kept some data about exits in contract. Please give it a second thought if it's still the case in our new design. I am not able to do that now. 
On the other hand, we can revert that change later, so do as you wish.

Reply:
There would be two kinds of "exit data", one in framework layer which is this one that only contains pointer for the framework to know which exit processor to call and some minimum info to use for "process exit". 

Another is the real exit data that contains all the exit info that would be stored in ExitGame contract.

For the interaction of SE and IFE, if conflicting with same output of same tx, that should be able to handled within the ExitGame contract as both data exists in the same contract. The one with issue would be SE conflicting IFE input because there is potential cross ExitGame contract data dependency. And that is why this issue: https://github.com/omisego/plasma-contracts/issues/110.

I prefer to solve the conflict by using "Vault" that flags finalized output instead of using exit mapping to represent the finalized output (as proposed in the issue above) since Vault is global in a way anyway and it makes sense that a vault knows what output has been finalized there.  This comes with the tradeoff of gas cost but make each ExitGame contract more independent and don't need to be aware of other contract implementation detail.